tmt/steps/prepare/artifact/providers/koji.py

```diff
@@ -184,3 +230,119 @@ def _download_artifact(
             )
         except Exception as error:
             raise DownloadError(f"Failed to download '{artifact}'.") from error
+
+
+@provides_artifact_provider("koji.task")  # type: ignore[arg-type]
+class KojiTask(KojiArtifactProvider):
+    # TODO: Make RPM_SCRATCH_BASE_URL configurable via FMF/CLI, not just env var
+    SCRATCH_BASE_URL = os.getenv(
+        "RPM_SCRATCH_BASE_URL", "https://kojipkgs.fedoraproject.org"
+    ).rstrip("/")  # For scratch builds
+
+    @cached_property
+    def build_id(self) -> Optional[int]:
+        task_id = int(self.id)
+        builds = self._call_api("listBuilds", taskID=task_id) or []
+        if builds:
+            build_id = builds[0]["build_id"]  # Assume the task produced a single build
+            assert isinstance(build_id, int)
+            return build_id
+        return None
+
+    def _get_task_children(self, task_id: int) -> list[int]:
+        """
+        Recursively fetch all child tasks of the given task ID.
+
+        :param task_id: The parent task ID
+        :return: List of all child task IDs
+        """
+        child_tasks: list[int] = [task_id]  # Include the parent task itself
+        direct_children = self._call_api("getTaskChildren", task_id) or []
+        for child in direct_children:
+            child_id = child["id"]
+            assert isinstance(child_id, int)
+            child_tasks.append(child_id)
+            # Recursively fetch grandchildren
+            child_tasks.extend(self._get_task_children(child_id))
+        return child_tasks
+
+    def _rpm_from_scratch(self, task_id: int, filename: str) -> RpmArtifactInfo:
+        task_dir = f"{task_id % 10000}/{task_id}"  # TODO: use pathInfo; signing off for now ;)
+        url = f"{self.SCRATCH_BASE_URL}/work/tasks/{task_dir}/{filename}"
+        try:
+            if filename.endswith('.src.rpm'):
+                base = filename[:-8]  # Remove '.src.rpm'
+                arch = 'src'
+            else:
+                base, arch, _ = filename.rsplit(".", 2)
+            name, version, release = base.rsplit("-", 2)
+        except ValueError:
+            raise ValueError(f"Invalid RPM filename format: '{filename}'")
```

Comment:
We do not need this part, only the `url`. Whether we will need to handle the `nvr` and `arch` we can handle when we know what we need it for. Best would be to `rpm -q` the files for the relevant information.

Reply:
Can be overwritten and pass the current name directly